### PR TITLE
Remove `ReturnTypeWillChange`

### PR DESCRIPTION
**Pull request by derrabus**
| Q             | A
| ------------- | ---
| Branch?       | 6.0
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | N/A
| License       | MIT
| Doc PR        | N/A

This PR removed the `ReturnTypeWillChange` attribute everywhere by adding actual return types.

### CODE REVIEW

**Comment by derrabus** [comment]
We should probably wait until #42379 has been merged up as well.

**Comment by Tobion** [comment]
#42379 has been merged. Please include the relevant changes here as well.

**Comment by derrabus** [comment]
Done.

File: src/Symfony/Component/Finder/Finder.php
Comment (Tobion) [inline src/Symfony/Component/Finder/Finder.php:606]: we might want to fix those phpdocs in lower branches?

File: src/Symfony/Component/Form/FormErrorIterator.php
Comment (Tobion) [inline src/Symfony/Component/Form/FormErrorIterator.php:151]: return annotation can be removed for this

**Review by Tobion — APPROVED** [review]

**Comment by derrabus** [comment]
The failing test is fixed and backported as #42384. I'll take care of the Psalm failures later.

File: src/Symfony/Component/DomCrawler/Crawler.php
Comment (azjezz) [inline src/Symfony/Component/DomCrawler/Crawler.php]:
```suggestion
     * @return \ArrayIterator<int, \DOMNode>
```

Comment (derrabus) [reply src/Symfony/Component/DomCrawler/Crawler.php]: Yeah, I know. As I said, I'm going to fix all those issues later.

Comment (derrabus) [reply src/Symfony/Component/DomCrawler/Crawler.php]: (later as in _before_ we merge, of course)

**Comment by derrabus** [comment]
PR is ready again.

File: src/Symfony/Component/Finder/Iterator/RecursiveDirectoryIterator.php
Comment (derrabus) [inline src/Symfony/Component/Finder/Iterator/RecursiveDirectoryIterator.php]: This change triggers the deprecation again. PHP expects the return type to be `RecursiveDirectoryIterator`, but that type is too narrow for our current implementation, unfortunately.

Comment (derrabus) [reply src/Symfony/Component/Finder/Iterator/RecursiveDirectoryIterator.php]: Filed as #42468. I'm reverting this change for now.

File: src/Symfony/Component/Form/FormBuilder.php
Comment (Tobion) [inline src/Symfony/Component/Form/FormBuilder.php:211]: are you gonna back-port all these type fixes or does it not matter? this is mainly to make psalm happy. but we do also check psalm on lower branches right.

Comment (derrabus) [reply src/Symfony/Component/Form/FormBuilder.php:211]: Yes, I intend to do that.

**Comment by Tobion** [comment]
Thank you @derrabus.

**Pull request merged**
